source/agora/script/Engine.d

```diff
@@ -945,6 +1103,175 @@ unittest
         null);
 }
 
+// OP.CHECK_SEQ_SIG / OP.VERIFY_SEQ_SIG
+unittest
+{
+    // Expected top to bottom on stack: <min_seq> <key> [<new_seq> <sig>]
+    //
+    // Unlock script pushes in order: [<sig>, <new_seq>]
+    // Lock script pushes in order: <key>, <min_seq>
+    //
+    // Stack:
+    //   <min_seq>
+    //   <key>
+    //   <new_seq>
+    //   <sig>
+
+    scope engine = new Engine(TestStackMaxTotalSize, TestStackMaxItemSize);
+    const Transaction tx = { inputs : [Input.init] };
+    test!("==")(engine.execute(
+        Lock(LockType.Script, [OP.CHECK_SEQ_SIG]), Unlock.init, tx, tx.inputs[0]),
+        "CHECK_SEQ_SIG opcode requires 4 items on the stack");
+    test!("==")(engine.execute(
+        Lock(LockType.Script, [OP.VERIFY_SEQ_SIG]), Unlock.init, tx, tx.inputs[0]),
+        "VERIFY_SEQ_SIG opcode requires 4 items on the stack");
+    test!("==")(engine.execute(
+        Lock(LockType.Script, [1, 42, 1, 42, 1, 42, OP.CHECK_SEQ_SIG]),
+        Unlock.init, tx, tx.inputs[0]),
+        "CHECK_SEQ_SIG opcode requires 4 items on the stack");
+    test!("==")(engine.execute(
+        Lock(LockType.Script,
+            [1, 42, 1, 42, 1, 42, 1, 42, OP.CHECK_SEQ_SIG]),
+        Unlock.init, tx, tx.inputs[0]),
+        "CHECK_SEQ_SIG opcode requires 8-byte minimum sequence on the stack");
+
+    const seq_0 = ulong(0);
+    const seq_1 = ulong(1);
+    const seq_0_bytes = nativeToLittleEndian(seq_0);
+    const seq_1_bytes = nativeToLittleEndian(seq_1);
+
+    test!("==")(engine.execute(
+        Lock(LockType.Script,
+              [ubyte(1), ubyte(1)]  // wrong pubkey size
+            ~ toPushOpcode(seq_0_bytes)
```

Comment:
We usually put the `~` operator at the end of a line, I guess.

Reply:
It depends. In some code it's done that way.

But recently I've realized that putting it on its own line makes it easier to see the concatenations. Because they all get aligned. For example:

```D
int[] arr = [1, 2, 3] ~
    [4, 5, 6, 7, 8 9, 10, 11, 12] ~
    [7, 8];
```

Compared to:

```D
int[] arr = [1, 2, 3]
    ~ [4, 5, 6, 7, 8 9, 10, 11, 12]
    ~ [7, 8];
```

In he second case the concatenations are all on the same column and easy to see. I think this is also the style Mathias prefers.

Reply:
Ah, I understand this is our preference.